mobile/library/cc/request_headers_builder.cc

```diff
@@ -43,6 +43,19 @@ RequestHeadersBuilder::addUpstreamHttpProtocol(UpstreamHttpProtocol upstream_htt
   return *this;
 }
 
+RequestHeadersBuilder&
+RequestHeadersBuilder::enableRequestCompression(CompressionAlgorithm algorithm) {
+  std::string value;
+  switch (algorithm) {
+  case CompressionAlgorithm::gzip:
+    value = "gzip";
+  case CompressionAlgorithm::brotli:
+    value = "brotli";
+  }
+  internalSet("x-envoy-mobile-compression", std::vector<std::string>{value});
```

Comment:
nit: Should we consider having a named constant for this value?

Reply:
Sure, where would that live ideally? Although I don't think it'd be worth plumbing it through to be accessed from other languages personally.

Reply:
Good question. I'm not sure. mostly I was thinking it would be nice to share here and in engine_builder.cc. But I'm happy to keep it as is.